shm/Game.cpp

```diff
@@ -100,19 +100,20 @@ void Game::travel() {
     system("clear");
     printCurrentPositionOnMap();        
     auto destination = map_->getIsland(getTravelLocation()); 
-    while (!destination || destination == currentPosition_) {
+    while (!destination || destination->getPosition() == currentPosition_->getPosition()) {
         if (!destination) {
             printPromptInvalidDestination();
         }
-        if (destination == currentPosition_) {
+        else if (destination->getPosition() == currentPosition_->getPosition()) {
             printPromptCurrentPositionEqualsDestination();
         }
         printCurrentPositionOnMap();
         destination = map_->getIsland(getTravelLocation());
     }
 
     auto distance = map_->getDistanceToIsland(destination);
-    map_->setCurrentPosition(destination);
+    currentPosition_ = std::make_unique<Island>(*destination);
+    map_->setCurrentPosition(currentPosition_.get());
```

Comment:
Why construct a new pointer from existing object raw pointer to just pass a raw pointer to function? I don't think this code is needed
```suggestion
    currentPosition_ = std::make_unique<Island>(*destination);
    map_->setCurrentPosition(currentPosition_.get());
```

Reply:
Also, upon leaving this function the smart pointer will destroy itself and we will probably be left with a dangling pointer. So it's unnecessary and dangerous